ledger/participant-integration-api/src/main/scala/platform/index/InMemoryStateUpdater.scala

```diff
@@ -44,12 +46,21 @@ final class InMemoryStateUpdater(
         Update.CommandRejected,
     ) => TransactionLogUpdate.TransactionRejected,
     updateLedgerEnd: (Offset, Long) => Unit,
+    updatePackageMetadata: Update.PublicPackageUpload => Unit,
 ) {
 
+  val packageMetadataFlow: Sink[(Vector[(Offset, Update)], Long), Future[Done]] = Sink.foreach {
+    case (batch, _) =>
+      batch
+        .collect { case (_, event: Update.PublicPackageUpload) => event }
+        .foreach(updatePackageMetadata)
+  }
+
   // TODO LLP: Considering directly returning this flow instead of the wrapper
-  def flow: UpdaterFlow =
+  val flow: UpdaterFlow =
     Flow[(Vector[(Offset, Update)], Long)]
       .filter(_._1.nonEmpty)
+      .alsoTo(packageMetadataFlow)
```

Comment:
AFAIK alsoTo will break consistency for us.
Our requirement is, that as ledger end is updated, we guarantee that the pacakge metadata is updated upto ledger end.
This alsoTo will start a computation which will indeed backpressure, but the same batch will flow through the main pipeline without any delay, which at the end bumps the ledger end, and announces the new ledger end for the streams (bumps the Dispatcher).
So these guys are racing.
But ledger end pumping must come after updating the packageMetadataView